dune/xt/functions/lambda/global-flux-function.hh

```diff
@@ -0,0 +1,145 @@
+// Authors:
+//   Felix Schindler (2013 - 2017)
+//   Rene Milk       (2013 - 2016)
+//   Sven Kaulmann   (2013)
+//   Tobias Leibner  (2014)
+
+#ifndef DUNE_XT_FUNCTIONS_LAMBDA_GLOBAL_FLUX_FUNCTION_HH
+#define DUNE_XT_FUNCTIONS_LAMBDA_GLOBAL_FLUX_FUNCTION_HH
+
+#include <functional>
+
+#include <dune/xt/common/memory.hh>
+
+#include <dune/xt/functions/interfaces.hh>
+
+namespace Dune {
+namespace XT {
+namespace Functions {
+
+
+template <class U, size_t state_derivative_order = 0, class R = typename U::RangeFieldType, size_t r = 1, size_t rC = 1>
+class GlobalLambdaFluxFunction
+{
+  static_assert(AlwaysFalse<U>::value, "Not available for these dimensions or this state_derivative_order!");
+};
+
+/**
+ * \brief A flux function given by a lambda expression which can be evaluated without an entity.
+ *
+ *        To model the flux F in a Burgers equation, F(u) := u^p, with a variable exponent p given a localizable
+ *        function u, use as
+ * \code
+typedef some_localizable_function U;
+typedef LocalLambdaFluxFunction<U> FluxType;
+
+FluxType F([](const typename F::DomainType& x,
+              const typename F::StateRangeType& u,
+              const XT::Common::Parameter& mu) { return std::pow(u[0], mu.get("power").at(0)); },
+           XT::Common::ParameterType("power", 1),
+           "burgers_flux",
+           [](const XT::Common::Parameter& mu) { return mu.get("power").at(0);});
+\endcode
+ *        The XT::Common::ParameterType provided on construction ensures that the XT::Common::Parameter mu which is
+ *        passed on to the lambda is of correct type.
+ *        The (optional) fourth argument is a lambda function that takes a parameter and returns the order of the
+ *        GlobalLambdaFluxFunction with this parameter. If you do no provide this order lambda, you can't call
+ *        the order() method, everything else will work as expected.
+ * \note  This function does not implement jacobians.
+ */
+template <class U, class R, size_t r, size_t rC>
+class GlobalLambdaFluxFunction<U, 0, R, r, rC>
+    : public GlobalFluxFunctionInterface<typename U::E, typename U::D, U::d, U, 0, R, r, rC>
+{
+  typedef GlobalFluxFunctionInterface<typename U::E, typename U::D, U::d, U, 0, R, r, rC> BaseType;
+
+public:
+  using typename BaseType::DomainType;
+  using typename BaseType::StateRangeType;
+  using typename BaseType::RangeType;
+
+private:
+  typedef std::function<RangeType(const DomainType&, const StateRangeType&, const Common::Parameter&)> LambdaType;
+  typedef std::function<size_t(const Common::Parameter&)> OrderLambdaType;
+
+public:
+  GlobalLambdaFluxFunction(LambdaType lambda,
+                           const Common::ParameterType& param_type,
+                           const std::string nm = "globallambdafunction",
+                           OrderLambdaType order_lambda =
+                               [](const Common::Parameter&) {
+                                 DUNE_THROW(NotImplemented, "");
```

Comment:
The error message could be a bit more verbose, as in "You have to provide an order lambda on construction", or something like that...